code/game/objects/items/tools/weldingtool.dm

```diff
@@ -115,7 +115,7 @@
 
 	if(affecting && affecting.status == BODYPART_ROBOTIC && user.a_intent != INTENT_HARM)
 		//only heal to 25 if limb is damaged to or past 25 brute, otherwise heal normally
-		var/difference = affecting.brute_dam - 25
+		var/difference = affecting.brute_dam - 0
```

Comment:
No. Remove the code entirely or keep it entirely, don't do a shitty half-measure like this.